### PR TITLE
Fix code-cov integration

### PR DESCRIPTION
**Pull request by lakpa**
Resolves [788](https://github.com/tuist/tuist/issues/788)

### CODE REVIEW

**Comment by vytis** [comment]
I think the problem is that this PR is coming from a fork - the secrets are not exposed in that case and all checks that need some auth tokens are failing. This is by design, otherwise you could just print or upload secrets by simply opening a PR in the repo. 

Try pushing a branch directly to the repo with the same changes and see how it goes.

**Comment by lakpa** [comment]
> I think the problem is that this PR is coming from a fork - the secrets are not exposed in that case and all checks that need some auth tokens are failing. This is by design, otherwise you could just print or upload secrets by simply opening a PR in the repo.
> 
> Try pushing a branch directly to the repo with the same changes and see how it goes.

code coverage report uploading fail at

`Please provide the repository token to upload reports via `-t :repository-token``

it appears as if following repository-token is needed @pepibumur

**Comment by vytis** [comment]
I can access it from https://codecov.io/gh/tuist/tuist/settings, you should be able to see it to I think!

**Comment by lakpa** [comment]
> I can access it from https://codecov.io/gh/tuist/tuist/settings, you should be able to see it to I think!

looking at the doc, it says `-t` set the `private repository token` (option) set environment variable CODECOV_TOKEN=:uuid

**Comment by vytis** [comment]
As far as I remember this is the only token that is there - probably as members with access to the repo we can see it. However it's better to set it as a secret variable in Actions, not as plaintext in the repo. This is where we need @pepibumur help, I don't see a way to do it.

**Comment by lakpa** [comment]
> > I can access it from https://codecov.io/gh/tuist/tuist/settings, you should be able to see it to I think!
> 
> looking at the doc, it says `-t` set the `private repository token` (option) set environment variable CODECOV_TOKEN=:uuid

It is not required for public repositories

**Comment by lakpa** [comment]
I found a workaround as suggested in the following reported issue https://github.com/codecov/codecov-node/issues/16

File: .github/workflows/tuist.yml [hunk 1]
```diff
@@ -10,12 +10,16 @@ jobs:
...
-        run: xcodebuild test -scheme tuist-Package -enableCodeCoverage YES
+        run: |
+          rm -rf .coverage
+          mkdir -p .coverage
+          swift test --enable-code-coverage
+      - name: Convert code coverage to lcov format
+        run: xcrun llvm-cov export -format="lcov" .build/debug/tuist -instr-profile .build/debug/codecov/default.profdata > ./.coverage/coverage.lcov
       - name: Send test coverage report
-        run: bash <(curl -s https://codecov.io/bash)
+        run: |
```
Comment (pepibumur) [inline .github/workflows/tuist.yml]: Add the following:

```yaml
env:
  CODECOV_TOKEN: ${{ secrets. CODECOV_TOKEN }}
```

And GitHub will expose the secret token to the step so codecov will use it automatically.

**Comment by lakpa** [comment]
Closing this PR as opened another one from master

**Pull request closed without merging**